Add clear query editors tab

shivtools opened the pull request:
**Summary:** It ends up being very time consuming to manually close all query editors. This PR adds the ability to clear all query editor tabs in SQL Lab. Addresses https://github.com/apache/incubator-superset/issues/6588 

**Test plan:** Jasmine test has been added. 
 
![cleareditors](https://user-images.githubusercontent.com/12538052/50745452-11788100-11df-11e9-95b2-24794dbc1cd2.gif)

Commit:
Add clear query editors tab

betodealmeida commented on superset/assets/src/SqlLab/components/TabbedSqlEditors.jsx:
```diff
@@ -145,6 +145,13 @@ class TabbedSqlEditors extends React.PureComponent {
     if (key === 'add_tab') {
       this.newQueryEditor();
+    } else if (key === 'close_all_tabs') {
+      if (window.confirm('Are you sure you wish to clear all query editors?')) {
```
Here it would be better to use the `<Dialog>` component, like you mentioned in Slack. You can see how it's used here: https://github.com/apache/incubator-superset/blob/8307716d723d7e16d3e5dd579a6d9e0f8e5413d8/superset/assets/src/datasource/DatasourceModal.jsx#L42

betodealmeida reviewed (changes requested):
Awesome, thanks for working on this @shivtools! I added a couple comments... like you said, we should use the `<Dialog>` component for consistency, but replacing the `window.confirm` code with it should be straightforward.

mistercrunch commented:
A related though that's been discussed before would be to change the status indicator (green circle) to become an `X` for closing on hover of the tab.

Also wanted to point out that the approach here is not common design-wise (though to be fair the tab caret/extension menu that we have now isn't as well...). We should follow the classic patterns as much as possible. That would mean matching Chrome's behavior to just have an `X` on the right.

shivtools commented:
Thanks to both of you for the review! @mistercrunch just to clarify, would your suggested changes (implementing an `X` on the right of each tab) be implemented as part of this PR? If it is to be part of this PR, then what would happen to the `Rename tab` and `Hide tool bar` options in the extension menu?

<img width="216" alt="screen shot 2019-01-07 at 10 35 59 pm" src="https://user-images.githubusercontent.com/12538052/50813889-a6a57380-12cc-11e9-8a0e-f49c138c2109.png">

Commit:
Change window.confirm to Dialog, fix breaking cypress sqllab test

shivtools commented:
I've incorporated @betodealmeida's suggested changes and fixed a breaking sqllab cypress test. The resulting change can be seen below: 

![superset](https://user-images.githubusercontent.com/12538052/50851126-beaede80-1330-11e9-94ca-2fee51d75fc4.gif)

shivtools commented:
Hey @mistercrunch and @betodealmeida, I think PR 6617 seems to use a preferred approach design wise. Let me know if you think I should close this PR :)
